### PR TITLE
Added condition to skip non-existent folders of nltk.data.path

### PR DESCRIPTION
**Pull request by Niveda21694**
Fixed issue with loading nltk packages. Kindly test and review.

### CODE REVIEW

**Comment by Niveda21694** [comment]
I'm curious to know why wasn't this approved?  This solution resolved error while converting Python scripts that has nltk packages imported to exe.

File: PyInstaller/loader/rthooks/pyi_rth__nltk.py [hunk 1]
```diff
@@ -12,5 +12,13 @@
 import os
 import nltk
 
+datas = []
+# loop through the data directories and add them
+for p in nltk.data.path: #environment variable of nltk
+	if(os.path.exists(p)):
```
Comment (htgoebel) [inline PyInstaller/loader/rthooks/pyi_rth__nltk.py]: Please follow PEP 8 when formatting.

**Comment by htgoebel** [comment]
Thank you for this pull-request.

How does this change relate to #3775?

Please reword the **commit messages** to comply to our [Commit Message Rules](https://pyinstaller.readthedocs.io/en/latest/development/commit-messages.html#commit-messages). You also need to submit a [**changelog entry**](https://pyinstaller.readthedocs.io/en/latest/development/changelog-entries.html) so our users can learn about your change. (This is a new requirement since last September to speed up releasing.)

When updating a pull-request, you can simply (force) push the updated branch to github again. This will automatically update the pull-request (which follows the branch, not the commit). So you do not need to close the pull-request and open a new one. This also has the benefit that the discussion history is kept. For detailed instructions please read [Updating a Pull-Request](https://pyinstaller.readthedocs.io/en/stable/development/pull-requests.html#updating-a-pull-request) in the manual.

**Comment by Niveda21694** [comment]
@htgoebel , done the requested changes. 
**How does this change relate to #3775?**
Answer: As a part of #3775 , the nltk files are loaded from the environment path "nltk.data.path" which is a list created by default when we import nltk and points to various locations in our system say, C:\, D:\ and E:\ drive. When the user system does not have any of the path in list, the program fails with error "nltk-data folder not found"

Added a condition to check if the path exists in user system before trying to load data.
Example:
`>>> nltk.data.path
['C:\\Users\\xxx/nltk_data', 'C:\\nltk_data', 'D:\\nltk_data', 'E:\\nltk_data
', 'C:\\Users\\xxx\\AppData\\Local\\Continuum\\anaconda3\\envs\\myenv\\nltk_d
ata', 'C:\\Users\\xxx\\AppData\\Local\\Continuum\\anaconda3\\envs\\myenv\\sha
re\\nltk_data', 'C:\\Users\\xxx\\AppData\\Local\\Continuum\\anaconda3\\envs\\
myenv\\lib\\nltk_data', 'C:\\Users\\xxx\\AppData\\Roaming\\nltk_data']`

The hook-nltk presumes to have all the listed paths in users machine which fails when it does not exsit.

**Comment by Niveda21694** [comment]
@htgoebel, please confirm if that answers your question or any more changes are required to merge this change.

**Comment by Rens0506** [comment]
Just to let you know i tried the fix that is requested to be pulled for the merge and the result was
that this line:

` 
+nltk.data.path.clear()
`


is a faulty line which gives the error message:
AttributeError: 'list' object has no attribute 'clear'

so either change it to:
nltk.data.path=[]
or
remove the line

**Comment by Niveda21694** [comment]
@Rens0506, done the change for replacing clear with del command.

**Comment by Rens0506** [comment]
@Niveda21694 , thx for considering my input.
Another thing that i could see is a follow up error due to the line:


`nltk.data.path.append(os.path.join(sys._MEIPASS, "nltk_data"))`

the error:
NameError: name 'sys' is not defined occurs.
and adding :

`import sys `
won"t solve the problem since 
the attribute:
`sys._MEIPASS` will be lost since  the issue:

> AttributeError: 'module' object has no attribute '_MEIPASS' (redux) #1785 
will appear

**Comment by Niveda21694** [comment]
@htgoebel , please share your knowledge on this.

**Pull request closed without merging**